tools/generate-interface.sh

```diff
@@ -0,0 +1,38 @@
+#! /bin/bash
+# generate-interface.sh generates the interface for the Payments contract
+
+set -euo pipefail
+
+OUTPUT_PATH=src/interfaces/IPayments.sol
+
+# Check for required tools
+for cmd in forge jq npx; do
+  if ! command -v $cmd &> /dev/null; then
+    echo "Error: '$cmd' is not installed or not in PATH." >&2
+    exit 1
+  fi
+done
+
+# Build the contracts
+echo "Building contracts..."
+forge build --force
+
+# Create dir if it doesn't exist
+mkdir -p "$(dirname "$OUTPUT_PATH")"
+
+# Extract ABI using jq
+echo "Generating interface at $OUTPUT_PATH..."
+jq '.abi' out/Payments.sol/Payments.json | npx abi-to-sol@0.8.0 IPayments --license MIT --solidity-version "^0.8.27" > "$OUTPUT_PATH"
```

Comment:
optional, but you could pull these two versions up to variables at the top of the file so we can update it more easily, but it's not a big deal, we'll probably only ever change this when stuff doesn't work the way we expect -- unless we have the solidity version hard-wired somewhere else too?